integreat_cms/static/src/js/pois/poi-map.ts

```diff
@@ -1,165 +1,242 @@
 
-    container.classList.remove("hidden");
+    // Allow to open the map in full screen
+    map.addControl(new FullscreenControl({ container: container }));
+
+    updateHelpText(coordinates);
+
+    return [map, marker];
+}
+
+// Function to update the map at the current coordinates
+function updateMap(map: Map, marker: Marker) {
+    // Get the coordinates from the input fields
+    const coordinates = getCoordinates();
+
+    if (coordinates) {
+        // If coordinates are given, fly to their location
+        marker.setLngLat(coordinates).addTo(map);
+        map.flyTo({
+            center: coordinates,
+            zoom: 15,
+        });
+    } else {
+        // Else, remove the marker from the map
+        marker.remove();
+        // And show a bounding box of the region
+        map.fitBounds(JSON.parse(map.getContainer().dataset.boundingBox) as LngLatBoundsLike);
+    }
+
+    updateHelpText(coordinates);
+}
+
+// Adjust help text below the map
+function updateHelpText(coordinates: LngLat) {
+    const setMapPositionText = document.getElementById("set_map_position_text");
+    const changeMapPositionText = document.getElementById("change_map_position_text");
+    if (coordinates) {
+        setMapPositionText.classList.add("hidden");
+        changeMapPositionText.classList.remove("hidden");
+    } else {
+        setMapPositionText.classList.remove("hidden");
+        changeMapPositionText.classList.add("hidden");
+    }
 }
 
-async function drag_and_drop(lngLat: maplibregl.LngLat) {
-    document.getElementById("drag_and_drop_change").classList.remove("hidden");
+// Fetch the address at a given location
+async function fetchAddress(marker: Marker) {
+    // Show loading spinner
+    document.getElementById("address_loading").classList.remove("hidden");
+    // Hide error in case it was shown before
+    document.getElementById("no_address_found").classList.add("hidden");
+
+    // Get the new coordinates of the marker
+    const coordinates = marker.getLngLat();
+
+    // Show new coordinates
+    document.getElementById("updated_latitude").textContent = coordinates.lat.toString();
+    document.getElementById("updated_longitude").textContent = coordinates.lng.toString();
+    document.getElementById("update_position_from_map_marker").classList.remove("hidden");
 
+    // Fetch the address at this coordinates
     const response = await fetch(document.getElementById("map").dataset.url, {
         method: "POST",
         headers: {
             "X-CSRFToken": getCsrfToken(),
         },
         body: JSON.stringify({
-            lng: lngLat.lng,
-            lat: lngLat.lat,
+            longitude: coordinates.lng,
+            latitude: coordinates.lat,
         }),
     });
 
-    const data = await response.json();
-
-    let error = document.getElementById("nominatim-error");
-    if (response.status != 200) {
-        error.textContent = data.error;
-        error.classList.remove("hidden");
-        return;
```

Comment:
I think we shouldn't abort on error - even if the address cannot be found, it might be useful to update only the coordinates without the address.